lib/dal/src/workspace_snapshot/node_weight.rs

```diff
@@ -457,6 +457,36 @@ impl NodeWeight {
         }
     }
 
+    /// Many node kinds need to have complete control of their outgoing edges
+    /// relative to another changeset in order to have a correctly constructed
+    /// graph. For example, only one set of children of a given attribute value
+    /// should "win" in a rebase operation, otherwise there could be duplicate
+    /// child values for an attribute value. This method will be called during
+    /// conflict detection in order to produce a conflict if the change set
+    /// being rebased has unseen edges of this type for a given "container"
+    /// node. If edge kinds are not returned here, those unseen edges will be
+    /// silently merged with the `onto` changeset's edges. This a "business"
+    /// logic problem, rather than a purely graph-theoretical one.
+    pub const fn exclusive_outgoing_edges(&self) -> &[EdgeWeightKindDiscriminants] {
```

Comment:
Mostly just an observation:

I think this fairly clearly points us down the path of having more granular (rather than generic) edge weight kinds. It would be very difficult to express that _only_ `Use` edges that point to a specific `ContentAddressDiscriminant` `ContentNodeWeight` should be "exclusive", and that any other `Use` edges are allowed to an arbitrary number. At the very least, any singleton relationship would need to have a unique `EdgeWeightKindDiscriminant` among all of the edge weight kinds used with a particular `NodeWeightDiscriminant`.

Reply:
That was 100% a thought I had while implementing this. We likely need more distinction among edge weight kinds